journal/journal-app.js

```diff
@@ -0,0 +1,147 @@
+window.addEventListener('DOMContentLoaded', init);
+
+// Main function that initializes every other function
+function init() {
+    const journalContainer = document.getElementById('journal-app');
+    const addButton = journalContainer.querySelector('.add-journal');
+    const overlayRef = document.getElementById('overlay');
+    const modalRef = document.getElementById('modal');
+
+    addButton.addEventListener('click', () => addJournalNew(journalContainer, false));
+
+    // When a user clicks "out" of the journal content/popup, it closes it
+    
+    overlayRef.addEventListener('click', () => {
+        overlayRef.style.display = 'none';
+        modalRef.style.display = 'none';
+    });
+
+    // Add saved journals from localStorage
+    addJournalNew(journalContainer, true);
+}
+
+// Display the new journal element to the page
+function addJournalNew(journalContainer, existing) {
+    
+    const journalList = getJournals();
+    let journalID = 0;
+
+    const journalTemplate = {
+        id: Math.floor(Math.random() * 2000000),
+        title: "",
+        documentation: "",
+        reflection: ""
+    };
+    
+    const modalRef = document.getElementById('modal');
+
+    if (existing) {
```

Comment:
could you please add a comment about the logic behind the `existing` flag

Reply:
I understand the logic that if the journal alr exists, then we load the list of journals. But why should the existing flag ever be true to pass in the function?

Reply:
Same feedback, little confused about `existing`.

Reply:
The whole gist of the `existing` flag is to check whether or not there are journals from `localStorage` and then display those onto the journal page. The flag is only `true` when we initially load or attempt to reload the journal page.

This was my approach to utilizing `localStorage` prior to learning to use a database like SQLite. If I didn't include the sections of the `existing` flag, then if we reload the page or come back to the page, you won't see the previously created journals displayed at all (however it is located in `localStorage`).